src/main/scala/amailp/intellij/robot/parser/RobotParser.scala

```diff
@@ -17,6 +18,7 @@ object RobotParser extends PsiParser {
         case TestCasesHeader | TasksHeader => parseTableItemsWith(parseTestCaseDefinition); Some(ast.TestCasesTable)
         case KeywordsHeader => parseTableItemsWith(parseKeywordDefinition); Some(ast.KeywordsTable)
         case VariablesHeader => parseTableItemsWith(parseVariableDefinition); Some(ast.VariablesTable)
+        case CommentsHeader => parseTableItemsWithSubParser(CommentsTableParser); Some(ast.CommentsTable)
```

Comment:
Being a very simple table I would suggest not to crate an instance `SubParser` class and rather  a simple `def` to be passed to `parseTableItemsWith()`